### ISSUE TITLE
Building this into a Chrome Extension - can't point to your script directly yet

### ISSUE THREAD

**Issue by siakaramalegos** [ausi/respimagelint#87]
Hi there - thanks for making such a great tool. I love it, but sadly it does not work on pages that block iframes. So I'm working on building it into a [Chrome Extension](https://github.com/siakaramalegos/resp-image-lint-extension) to try to get around this. Long term, it might be nice to have it there anyway for folks confused by bookmarklets.

I have not figured out the x-frame-options header issue yet, but assuming that eventually happens, another issue is I can't point to your script directly yet. This is because you're querying the document for a script tag so get the base URL. For now, I've pasted it directly there: https://github.com/siakaramalegos/resp-image-lint-extension/blob/main/RespImageLint.js#L1007C191-L1007C245

The idea is to use the same base code and point to your stuff for sponsoring, etc. This is just a convenience wrapper.

**Comment by siakaramalegos** [comment]
I guess the question is, first, how do you feel about this? And secondly, if you like the idea, do you have a way you'd prefer to handle the baseURL issue that is easy for both of the bookmarklet and chrome extension to maintain?

**Comment by peter-neumann-dev** [comment]
Hi @siakaramalegos, I have already made a Chrome extension for myself as a workaround because I use a browser (Arc) that does not support bookmarklets yet. It is available in the chrome web store. You can check it out here: https://github.com/peter-neumann-dev/responsive-image-linter

For now, I kept the script just forked and changed the baseURL stuff as well. If the extension retrieves more updates frequently again, there should be a better way to keep it sync.

More here as well: https://github.com/ausi/respimagelint/issues/84

**Comment by ausi** [comment]
> how do you feel about this?

It’s great to see people using it. And I’m a little embarrassed that it works so unstable due to it being a bookmarklet. I still plan to transform it into a dev tools extension, but that might take a little longer still 😬 
Having browser extensions from @siakaramalegos and @peter-neumann-dev to fill that gap in the meantime is totally fine with me. We should probably also link to them in the “Usage” section of the readme, wdyt?

> … I can't point to your script directly yet. This is because you're querying the document for a script tag to get the base URL.

Would it help if I implement a fallback like this?

```js
const scriptBase = script ? script.src.split('?')[0].replace(/[^/]+$/, '') : 'https://ausi.github.io/respimagelint/';
```

**Comment by siakaramalegos** [comment]
I got it working!! https://github.com/siakaramalegos/resp-image-lint-extension

@peter-neumann-dev do you prefer to merge in my code or maintain a separate extension? Yours has a more complex file structure, so I'm guessing you'd want to structure it differently than I did. Let me know what you're thinking either way!

@ausi if you can make that tiny script change, it will be easier for us to maintain extensions too.

**Comment by siakaramalegos** [comment]
Unrelated - noticed this warning in the extension logs in case you wanted to optimize the code slightly @ausi 

> Canvas2D: Multiple readback operations using getImageData are faster with the willReadFrequently attribute set to true. See: https://html.spec.whatwg.org/multipage/canvas.html#concept-canvas-will-read-frequently

**Comment by peter-neumann-dev** [comment]
@siakaramalegos Nice! 🎉  I think the structure in my extension is looking more complex than it looks at a first glance. As I can see from your commit, it seems we only have to adjust two files:

- the `manifest.json` (https://github.com/peter-neumann-dev/responsive-image-linter/blob/main/src/manifest.json)
- the `service-worker.js` (https://github.com/peter-neumann-dev/responsive-image-linter/blob/main/src/service-worker.js) seems very similar to your `background.js`

If you like, you can create a pull request, or I'll add the same changes in mine extension. Do you have an example page where I can test the previous not working linting?

**Comment by siakaramalegos** [comment]
Yes, any Shopify store, but for convience here's our blog that we also built on shopify: https://performance.shopify.com/

**Comment by peter-neumann-dev** [comment]
Yes, I can reproduce it. Will you create a PR, or should I add the changes myself @siakaramalegos?

**Comment by ausi** [comment]
> if you can make that tiny script change

Done in f2002aa74fbb6d37ef59a55b835c6f1adec5b538

> noticed this warning in the extension logs in case you wanted to optimize the code slightly

Done in 134a5ae706b1f479ee8068cc8ec24463c0ca5c1d

**Comment by ausi** [comment]
> We should probably also link to them in the “Usage” section of the readme, wdyt?

I added a link to Peters extension now: https://github.com/ausi/respimagelint#usage

@siakaramalegos I’d also add a link to your extension once finished. Can you notify me then?

**Comment by siakaramalegos** [comment]
Sure! Though I gave Peter a PR so it looks like it will just be one
extension once we get it all ready. I'm so excited!

**Comment by peter-neumann-dev** [comment]
I have released a [new version ](https://github.com/peter-neumann-dev/responsive-image-linter/releases/tag/v1.1.0)of the extension and submitted the update to the Chrome Web Store for review. Once it is approved, the new version will be available automatically through the store. 🤞🏻😃

**Comment by peter-neumann-dev** [comment]
It is now published finally. After updating, you have to reactivate and accept to the new permission that it can block content. Afterward, it is working everywhere in my tests 👍🏻 🙂
